console/v1/types_console_sample.go

```diff
@@ -0,0 +1,183 @@
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata"`
+
+	// +required
+	Spec ConsoleSampleSpec `json:"spec"`
+}
+
+// ConsoleSampleSpec is the desired sample for the web console.
+// Samples will appear with their title, descriptions and a badge in a samples catalog.
+type ConsoleSampleSpec struct {
+	// title is the display name of the sample.
+	//
+	// It is required and must be no more than 50 characters in length.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	Title string `json:"title"`
+
+	// abstract is a short teaser the sample.
+	//
+	// It is required and must be no more than 100 characters in length.
+	//
+	// Currently, the abstract is shown on the sample card tile below the title and provider
+	// and is limited to three lines of content.
+	// +required
+	// +kubebuilder:validation:MaxLength=100
+	Abstract string `json:"abstract"`
+
+	// description of the sample.
+	//
+	// It is required and can have a maximum length of **4096** characters.
+	//
+	// It is a README.md-like content for additional information, links, pre-conditions, and other instructions.
+	// It will be rendered as Markdown so that it can contain line breaks, links, and other simple formatting.
+	// +required
+	// +kubebuilder:validation:MaxLength=4096
+	Description string `json:"description"`
+
+	// icon is an optional base64 encoded image and shown beside the sample title.
+	//
+	// The format must follow the [data: URL format].
+	//
+	//   data:[<mediatype>][;base64],<base64 encoded image>
+	//
+	// For example:
+	//
+	//   data:image;base64,             plus the base64 encoded image.
+	//
+	// Vector images can also be used. SVG icons must start with:
+	//
+	//   data:image/svg+xml;base64,     plus the base64 encoded SVG image.
+	//
+	// All sample catalog icons will be shown on a white background (also when the dark theme is used).
+	// The web console ensures that different aspect radios work correctly.
+	// Currently, the surface of the icon is at most 40x100px.
+	//
+	// [data: URL format]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs
+	// +optional
+	// +kubebuilder:validation:Pattern=`^data:`
+	Icon string `json:"icon"`
+
+	// type is an optional label to group multiple samples.
+	//
+	// It is optional and must be no more than 20 characters in length.
+	//
+	// Recommendation is a singular term like "Builder Image", "Devfile" or "Serverless Function".
+	//
+	// Currently, the type is shown a badge on the sample card tile in the top right corner.
+	// +optional
+	// +kubebuilder:validation:MaxLength=20
+	Type string `json:"type"`
+
+	// provider is an optional label to honor who provides the sample.
+	//
+	// It is optional and must be no more than 20 characters in length.
```

Comment:
Is 20 a bit limiting? Perhaps 50 would be better or is that likely to cause issues? Do you have some data that would show the average length of a provider from the various options for samples?

Reply:
How will you assess feedback that the length is too short? Is there any data we can use to inform this decision?

Reply:
Changed it to 50. For me it's up to the developer to check if their sample look and work correctly in the console. It will work also with 50 characters and looks "okay":

![image](https://github.com/openshift/api/assets/139310/66d57b57-7a70-47be-be74-e364390d2fa4)